addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -4417,22 +4417,17 @@ var SnippetsMenu = Widget.extend({
      * Preview on mobile.
      */
     _onMobilePreviewClick() {
-        this.trigger_up('request_mobile_preview');
-
-        // TODO refactor things to make this more understandable -> on mobile
-        // edition, update the UI. But to do it properly and inside the mutex
-        // this simulates what happens when a snippet option is used.
+        this._execWithLoadingEffect(async () => this.trigger_up('request_mobile_preview'), false);
         this._execWithLoadingEffect(async () => {
-            // TODO needed so that mobile edition is considered before updating
-            // the UI but this is clearly random. The trigger_up above should
-            // properly await for the rerender somehow.
-            await new Promise(resolve => setTimeout(resolve));
-
-            return new Promise(resolve => {
+            // TODO: the trigger_up above should properly await for the
+            // rerender somehow.
```

Comment:
Is there a re-render? Is not it just the iframe being resized? As said above, if there is another call to `_execWithLoadingEffect` that can be done by the mobile switch then the following code should be done afterwards, not synchronously.

Reply:
Hmm, I think you're right, it's not a rerender. But as said below, the code still has to wait for everything mobile-related to kick in before applying `snippet_option_update`.